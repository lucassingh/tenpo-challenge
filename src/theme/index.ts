import light from './lightTheme';

export default { light }