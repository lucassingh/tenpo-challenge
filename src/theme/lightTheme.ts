export default {
    colors: {
        primary: "#D4F9F5",
        secondary: "#00BAA4",
        background: "#F2F2F2",
        lightGray: "#F2F2F2",
        light: "#FFFFFF",
        dark: "#1C1C1C"
    }
}