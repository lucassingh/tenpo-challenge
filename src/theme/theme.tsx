import React from "react";
import { ThemeProvider } from "styled-components/native";
import theme from ".";

type Props = {
    children: React.ReactNode
}

export function Theme({children}: Props) {
    return <ThemeProvider theme={theme.light}>
        {children}
    </ThemeProvider>
}