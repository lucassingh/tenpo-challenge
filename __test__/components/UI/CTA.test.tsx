import React from "react";
import 'jest-styled-components'
import styled from 'styled-components/native';
import theme from "../../../src/theme";
import { render } from '@testing-library/react-native'

describe('Test Container and text in component in <CTA />', () => {

    const Container = styled.View`
        width: 100%;
        height: 15%;
        background-color: ${theme.light.colors.primary};
        border-top-right-radius: 35px;
        border-top-left-radius: 35px;
        display: flex;
        padding-top: 5%;
        justify-content:start;
        align-items: center;
    `

    it('Should match with snapshot', () => {

        const container = render(<Container />);

        expect(container).toMatchSnapshot()
    })

    const Text = styled.Text`
        font-size: 18px;
        font-weight: 500;
        width: 100%;
        color: ${theme.light.colors.secondary};
    `

    it('Should match with snapshot', () => {

        const container = render(<Text />);

        expect(container).toMatchSnapshot()
    })
})