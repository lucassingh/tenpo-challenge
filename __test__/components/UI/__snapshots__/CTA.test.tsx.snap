// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Container and text in component in <CTA /> Should match with snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#D4F9F5",
        "borderTopLeftRadius": 35,
        "borderTopRightRadius": 35,
        "display": "flex",
        "height": "15%",
        "justifyContent": "start",
        "paddingTop": "5%",
        "width": "100%",
      },
    ]
  }
/>
`;

exports[`Test Container and text in component in <CTA /> Should match with snapshot 2`] = `
<Text
  style={
    [
      {
        "color": "#00BAA4",
        "fontSize": 18,
        "fontWeight": "500",
        "width": "100%",
      },
    ]
  }
/>
`;
